SimTKcommon/Geometry/src/DecorativeGeometryRep.h

```diff
@@ -583,6 +583,174 @@ const DecorativeMeshFile& getMyMeshFileHandle() const {
 }
 };
 
+
+////////////////////////
+// DecorativeTorusRep //
+////////////////////////
+
+class DecorativeTorusRep : public DecorativeGeometryRep {
+public:
+    // no default constructor
+    explicit DecorativeTorusRep(Real torusR, Real tubeR) : torusRadius(torusR), tubeRadius(tubeR) {
+    }
+
+    const Real& getTorusRadius() const {
+        return  torusRadius;
+    }
+
+    const Real& getTubeRadius() const {
+        return  tubeRadius;
+    }
+
+    void setTorusRadius(Real torusR) {
+        torusRadius = torusR;
+    }
+
+    void setTubeRadius(Real tubeR) {
+        tubeRadius = tubeR;
+    }
+    // virtuals
+    DecorativeGeometryRep* cloneDecorativeGeometryRep() const {
+        DecorativeTorusRep* DGRep = new DecorativeTorusRep(*this);
+        return DGRep;
+    }
+
+    void implementGeometry(DecorativeGeometryImplementation& geometry) const {
+        geometry.implementTorusGeometry(getMyTorusHandle());
+    }
+
+    SimTK_DOWNCAST(DecorativeTorusRep, DecorativeGeometryRep);
+private:
+    Real torusRadius;
+    Real tubeRadius;
+
+    // This is just a static downcast since the DecorativeGeometry handle class is not virtual.
+
+    const DecorativeTorus& getMyTorusHandle() const {
+        return *static_cast<const DecorativeTorus*>(myHandle);
+    }
+};
+
+
+///////////////////////
+// DecorativArrowRep //
+///////////////////////
+
+class DecorativeArrowRep : public DecorativeGeometryRep {
+public:
+    // no default constructor
+    explicit DecorativeArrowRep(const Vec3& start, const Vec3& end, Real& tipLen) : 
+        start_point(start), end_point(end), tip_length(tipLen) {
+    }
+
+    const Vec3& getStartPoint() const {
+        return  start_point;
+    }
+
+    const Vec3& getEndPoint() const {
+        return  end_point;
+    }
+
+    const Real& getTipLength() const {
+        return  tip_length;
+    }
+
+    void setStartPoint(const Vec3& start) {
+        start_point = start;
+    }
+    void setEndPoint(const Vec3& end) {
+        end_point = end;
+    }
+    void setTipLength(Real& tlen) {
+        tip_length = tlen;
+    }
+
+    // virtuals
+    DecorativeGeometryRep* cloneDecorativeGeometryRep() const {
+        DecorativeArrowRep* DGRep = new DecorativeArrowRep(*this);
+        return DGRep;
+    }
+
+    void implementGeometry(DecorativeGeometryImplementation& geometry) const {
+        geometry.implementArrowGeometry(getMyArrowHandle());
+    }
+
+    SimTK_DOWNCAST(DecorativeArrowRep, DecorativeGeometryRep);
+private:
+    Vec3 start_point;
+    Vec3 end_point;
+    Real tip_length;
```

Comment:
These don't match the [coding conventions for variable names](https://github.com/simbody/simbody/blob/master/CONTRIBUTING.md#variables). Should be `startPoint` etc. (could also be `m_startPoint` etc. if you want to separate out data members, but otherwise no underscores).